scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -1012,16 +1012,17 @@
 (deftest test-receive-offers
   (let [declined-offer-ids-atom (atom [])
         offers-chan (async/chan (async/buffer 1))
+        match-trigger-chan (async/chan (async/sliding-buffer 1))
         mock-driver (reify msched/SchedulerDriver
                       (decline-offer [driver id]
                         (swap! declined-offer-ids-atom conj id)))
         offer-1 {:id {:value "foo"}}
         offer-2 {:id {:value "bar"}}
         offer-3 {:id {:value "baz"}}]
     (testing "offer chan overflow"
-      (sched/receive-offers offers-chan mock-driver [offer-1])
-      @(sched/receive-offers offers-chan mock-driver [offer-2])
-      @(sched/receive-offers offers-chan mock-driver [offer-3])
+      (sched/receive-offers offers-chan match-trigger-chan mock-driver [offer-1])
+      @(sched/receive-offers offers-chan match-trigger-chan mock-driver [offer-2])
+      @(sched/receive-offers offers-chan match-trigger-chan mock-driver [offer-3])
       (is (= @declined-offer-ids-atom [(:id offer-2) (:id offer-3)])))))
```

Comment:
we should also verify the data in `match-trigger-chan`. Possibly make it a larger chan than sliding buffer to match the exact number of trigger calls.